components/eam/src/physics/cam/ndrop.F90

```diff
@@ -1426,98 +1426,89 @@ subroutine ccncalc(state, pbuf, cs, ccn)
 
    ! Ghan et al., Atmos. Res., 1993, 198-221.
 
-   ! arguments
-
+  ! input arguments
    type(physics_state), target, intent(in)    :: state
    type(physics_buffer_desc),   pointer       :: pbuf(:)
+   real(r8), intent(in)  :: cs(pcols,pver)       ! air density [kg/m3]
 
-
-   real(r8), intent(in)  :: cs(pcols,pver)       ! air density (kg/m3)
-   real(r8), intent(out) :: ccn(pcols,pver,psat) ! number conc of aerosols activated at supersat (#/m3)
+  ! output arguments
+   real(r8), intent(out) :: ccn(pcols,pver,psat) ! number conc of aerosols activated at supersat [#/m3]
 
    ! local
 
    integer :: lchnk ! chunk index
    integer :: ncol  ! number of columns
-   real(r8), pointer :: tair(:,:)     ! air temperature (K)
-
-   real(r8) naerosol(pcols) ! interstit+activated aerosol number conc (/m3)
-   real(r8) vaerosol(pcols) ! interstit+activated aerosol volume conc (m3/m3)
-
-   real(r8) amcube(pcols)
-   real(r8) super(psat) ! supersaturation
-   real(r8), allocatable :: amcubecoef(:)
-   real(r8), allocatable :: argfactor(:)
-   real(r8) :: surften       ! surface tension of water w/respect to air (N/m)
-   real(r8) surften_coef
-   real(r8) a(pcols) ! surface tension parameter
-   real(r8) hygro(pcols)  ! aerosol hygroscopicity
-   real(r8) sm(pcols)  ! critical supersaturation at mode radius
-   real(r8) arg(pcols)
-   !     mathematical constants
-   real(r8) twothird,sq2
-   integer l,m,n,i,k
-   real(r8) log,cc
-   real(r8) smcoefcoef,smcoef(pcols)
+   real(r8), pointer :: tair(:,:)     ! air temperature [K]
+   real(r8) naerosol(pcols) ! interstit+activated aerosol number conc [#/m3]
+   real(r8) vaerosol(pcols) ! interstit+activated aerosol volume conc [m3/m3]
+   real(r8) amcube(pcols)  ! [m3]
+   real(r8) super(psat) ! supersaturation [fraction]
+   real(r8) :: amcubecoef(ntot_amode) ! [dimensionless]
+   real(r8) :: argfactor(ntot_amode)  ! [dimensionless]
+   real(r8) surften_coef  ! [m-K]
+   real(r8) aparam(pcols) ! surface tension parameter  [m]
+   real(r8) hygro(pcols)  ! aerosol hygroscopicity [dimensionless]
+   real(r8) sm(pcols)  ! critical supersaturation at mode radius [fraction]
+   real(r8) arg_erf_ccn(pcols) ! [dimensionless] 
+   integer lsat,imode,icol,kk
+   real(r8) smcoefcoef   ! [dimensionless]
+   real(r8) smcoef(pcols)  ! [m^(3/2)]
    integer phase ! phase of aerosol
+   !     mathematical constants
+   real(r8) pct_to_fraction, per_m3_to_per_cm3
```

Comment:
is it worth considering `percent_to_fraction`?

Reply:
Yes, this has been done.